src/Umbraco.Web/Templates/LocalLinkParser.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Umbraco.Core;
+using Umbraco.Core.Logging;
+using Umbraco.Web.PublishedCache;
+using Umbraco.Web.Routing;
+
+namespace Umbraco.Web.Templates
+{
+    /// <summary>
+    /// Utility class used to parse internal links
+    /// </summary>
+    public sealed class LocalLinkParser
```

Comment:
I really don't like this name, when it is not related to `Umbraco.Core.Deploy.ILocalLinkParser`. 
Can we find an alternative even that it actually parses local links.. 😕